src/telemetry/wifi_scan.rs

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::DeviceManagerError;
+use serde::Serialize;
+use wifiscanner;
+use wifiscanner::Wifi;
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct WifiScanResult {
+    channel: i32,
+    connected: bool,
+    essid: String,
+    mac_address: String,
+    rssi: i32,
+}
+
+pub fn get_wifi_scan_results() -> Result<Vec<WifiScanResult>, DeviceManagerError> {
+    let mut ret = Vec::new();
+    if let Ok(wifi_array) = wifiscanner::scan() {
+        for wifi in wifi_array {
+            let w = get_struct_from_wifi(wifi)?;
+            ret.push(w);
+        }
+    }
+
+    Ok(ret)
+}
+
+fn get_struct_from_wifi(wifi: Wifi) -> Result<WifiScanResult, DeviceManagerError> {
```

Comment:
Use `From trait` for converting the `wifi` type.
``` rust
impl TryFrom<Wifi> for WifiScanResult {
type Error = DeviceManagerError;
    fn try_from(value: AstarteType) -> Result<Self, Self::Error> {
    ....
    }
 }
 ```   
 `get_struct_from_wifi(wifi)?; -> wifi.try_into();`